src/pgduckdb_hooks.cpp

```diff
@@ -67,38 +96,41 @@ ContainsDuckdbFunctions(Node *node, void *context) {
 	}
 
 #if PG_VERSION_NUM >= 160000
-	return expression_tree_walker(node, ContainsDuckdbFunctions, context);
+	return expression_tree_walker(node, ContainsDuckdbItems, context);
 #else
-	return expression_tree_walker(node, (bool (*)()) ((void *) ContainsDuckdbFunctions), context);
+	return expression_tree_walker(node, (bool (*)())((void *)ContainsDuckdbItems), context);
 #endif
 }
 
 static bool
 NeedsDuckdbExecution(Query *query) {
-#if PG_VERSION_NUM >= 160000
-	return query_tree_walker(query, ContainsDuckdbFunctions, NULL, 0);
-#else
-	return query_tree_walker(query, (bool (*)()) ((void *) ContainsDuckdbFunctions), NULL, 0);
-#endif
+	return ContainsDuckdbItems((Node *)query, NULL);
 }
 
 static bool
-IsAllowedStatement(Query *query) {
+IsAllowedStatement(Query *query, bool throw_error) {
```

Comment:
nit- we could have kept `throw_error = false` by default? (would reduce the diff/simplify API?)